modules/internal/ChapelDistribution.chpl

```diff
@@ -251,8 +266,13 @@ module ChapelDistribution {
         if locking then
           _arrsLock.lock();
         _arrs_containing_dom += 1;
-        if addToList && trackArrays() then
-          _arrs.add(x);
+        if addToList && trackArrays() {
```

Comment:
IMO it should have `x.prev = nil` and `x.next = nil` here before the conditional to make sure they are cleared.

Reply:
Would an assert that x.prev/x.next are nil work for you? I don't think this will ever be called for anything but newly created arrays

Reply:
yes